HW2/P3/mandelbrot.pyx

```diff
@@ -32,15 +40,76 @@ cpdef mandelbrot(np.complex64_t [:, :] in_coords,
     assert in_coords.shape[1] == out_counts.shape[1],  "Input and output arrays must be the same size"
 
     with nogil:
-        for i in range(in_coords.shape[0]):
-            for j in range(in_coords.shape[1]):
-                c = in_coords[i, j]
-                z = 0
+
+        for i in prange(in_coords.shape[0], schedule='static', chunksize=1, num_threads=num_threads):
+            
+            # we want every eighth number
+            for j in xrange(0, in_coords.shape[1], 8):
+                # reset iteration counts
+                iter_counts = AVX.float_to_float8(1.0)
+
+                # we're computing values eight at a time, so we create a float8
+                # with the current and next 7 values. one float8 will hold the
+                # 8 real parts, and one will hold the 8 imaginary parts
+                
+                c_real = AVX.make_float8(in_coords[i, (j + 7)].real,
+                                         in_coords[i, (j + 6)].real,
+                                         in_coords[i, (j + 5)].real,
+                                         in_coords[i, (j + 4)].real,
+                                         in_coords[i, (j + 3)].real,
+                                         in_coords[i, (j + 2)].real,
+                                         in_coords[i, (j + 1)].real,
+                                         in_coords[i, (j + 0)].real)
+
+                c_imag = AVX.make_float8(in_coords[i, (j + 7)].imag,
+                                         in_coords[i, (j + 6)].imag,
+                                         in_coords[i, (j + 5)].imag,
+                                         in_coords[i, (j + 4)].imag,
+                                         in_coords[i, (j + 3)].imag,
+                                         in_coords[i, (j + 2)].imag,
+                                         in_coords[i, (j + 1)].imag,
+                                         in_coords[i, (j + 0)].imag)
+
```

Comment:
little stylistic nit - lots of extra blank lines here are generally not desirable.